### PR TITLE
fix: add expiresAt field

### PR DESCRIPTION
**Pull request by bodinsamuel**
Add field that log when package should be recrawled.
The logic is basic for now:
- 30 days for every packages
- 7 days if it is popular

The goal in a futur PR is to use this facet to easily query outdated package.
Because we can not query with SQL logic like "packages < DATE", the easiest will be to query the facets values, iterate on the values and filter one by one each value.

### CODE REVIEW

File: src/@types/pkg.ts [hunk 1]
```diff
@@ -72,6 +72,7 @@ export interface RawPkg {
   types: TsType;
   moduleTypes: ModuleType[];
   lastCrawl: string;
+  expiresAt: string;
```
Comment (Haroenv) [inline src/@types/pkg.ts:75]: we can put it in _searchInternal to avoid exposing it too much?

File: src/npm/index.ts [hunk 2]
```diff
@@ -283,6 +286,11 @@ async function getDownloads(pkgs: Array<Pick<RawPkg, 'name'>>): Promise<
       humanDownloadsLast30Days: numeral(downloadsLast30Days).format('0.[0]a'),
       downloadsRatio,
       popular,
+      expiresAt: popular
+        ? new Date(Date.now() + config.populareExpiresAt)
```
Comment (Haroenv) [inline src/npm/index.ts:290]:
```suggestion
        ? new Date(Date.now() + config.popularExpiresAt)
```

File: src/config.ts [hunk 3]
```diff
@@ -164,6 +166,8 @@ export const config = {
   indexSettings,
   indexSynonyms,
   indexRules,
+  expiresAt: 30 * 24 * 3600 * 1000,
+  populareExpiresAt: 7 * 24 * 3600 * 1000,
```
Comment (Haroenv) [inline src/config.ts:170]:
```suggestion
  popularExpiresAt: 7 * 24 * 3600 * 1000,
```

Comment (Haroenv) [inline src/config.ts:170]: these can be configured with `ms`, for consistency with the rest :)

**Review by Haroenv — COMMENTED** [review]
great idea!

File: src/config.ts (hunk 3 above)
Comment (bodinsamuel) [reply src/config.ts:170]: good point 👌🏻

File: src/@types/pkg.ts (hunk 1 above)
Comment (bodinsamuel) [reply src/@types/pkg.ts:75]: 👌🏻

**Review by Haroenv — APPROVED** [review]

File: src/@types/pkg.ts [hunk 4]
```diff
@@ -75,6 +75,7 @@ export interface RawPkg {
   moduleTypes: ModuleType[];
   lastCrawl: string;
   _searchInternal: {
+    expiresAt: string;
```
Comment (Haroenv) [inline src/@types/pkg.ts:78]: if we want to use it for a filter, why isn't it a timestamp/number?

Comment (bodinsamuel) [reply src/@types/pkg.ts:78]: same same, it's just easier to read as string no?

**Pull request merged**